File: Java/MedianOfTwoSortedArrays_4.java
import java.util.Arrays;
import java.util.Scanner;
import java.util.stream.IntStream;

class Solution_4 {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int nums[] =IntStream.concat(Arrays.stream(nums1),Arrays.stream(nums2)).toArray();
        int l =nums.length;
        Arrays.sort(nums);
        return (l%2==0) ? (nums[l/2] + nums[(l/2)-1])/2.0 : nums[l/2] ;
    }
}

class MedianOfTwoSortedArrays_4{
    public static void main(String [] args){
        Solution_4 obj =new Solution_4();
        Scanner sc=new Scanner(System.in);

        System.out.print("Number of Test cases: ");
        int n =sc.nextInt();
        sc.nextLine();

        for (int i = 0; i < n; i++) {
            System.out.print("Enter length of 2 arrays: ");
            int m = sc.nextInt();
            int n1 = sc.nextInt();
            sc.nextLine();

            System.out.println("Enter the elements of 1st array: ");
            int[] nums1 = new int[m];
            for (int j = 0; j < m; j++) {
                nums1[j] = sc.nextInt();
            }
            sc.nextLine();

            System.out.println("Enter the elements of 2nd array: ");
            int[] nums2 = new int[n1];
            for (int j = 0; j < n1; j++) {
                nums2[j] = sc.nextInt();
            }
            sc.nextLine();
            System.out.println("Median of 2 sorted arrays: " + obj.findMedianSortedArrays(nums1, nums2));
        }
    }
}

File: Python/MedianOfTwoSortedArrays_4.py
import statistics
class Solution:
    def findMedianSortedArrays(self, nums1: list[int], nums2: list[int]) -> float:
        nums=nums1+nums2
        return statistics.median(nums)
        
        
s=Solution()
n=int(input("Enter number of test cases : "))

for i in range(n):
    m,n=map(int,input("Enter length of 2 arrays : ").split())
    print("Enter the elements of 1st array : ")
    nums1=list(map(int,input(f"Enter {m} elements with space seperation :").split()))
    nums2=list(map(int,input(f"Enter {n} elements with space seperation :").split()))
    print(f"Mediam of 2 sorted arrays : {s.findMedianSortedArrays(sorted(nums1),sorted(nums2))}")